app/models/maintenance_stats/queries/github/releases_query.rb

```diff
@@ -0,0 +1,81 @@
+module MaintenanceStats
+  module Queries
+    module Github
+      class RepoReleasesQuery < BaseQuery
+        RELEASES_QUERY = Rails.application.config.graphql.client.parse <<-GRAPHQL
+          query($owner: String!, $repo_name: String!, $cursor: String){
+            repository(owner: $owner, name: $repo_name){
+              releases(first: 100, after: $cursor, orderBy: {field: CREATED_AT, direction: DESC}){
+                nodes {
+                  name
+                  publishedAt
+                }
+                totalCount
+                pageInfo {
+                  hasPreviousPage
+                  hasNextPage
+                  endCursor
+                  startCursor
+                }
+              }
+            }
+          }
+        GRAPHQL
+
+        VALID_PARAMS = [:owner, :repo_name, :end_date]
+        REQUIRED_PARAMS = [:owner, :repo_name]
+
+        def self.client_type
+          :v4
+        end
+
+        def query(params: {})
+          validate_params(params)
+
+          end_date = params[:end_date]
+          params = params.slice!(:end_date)
+
+          @releases = nil
+
+          page_info = { has_next_page: true, cursor: nil}
+          while(page_info[:has_next_page])
+            params[:cursor] = page_info[:cursor] if page_info[:cursor]
+            result = @client.query(RELEASES_QUERY, variables: params)
+
+            if check_for_errors(result)
+              @releases = result if @releases.nil?
```

Comment:
Is `return releases || result if check_for_errors(result)` too much?

Reply:
I think that's essentially the same statement.

Reply:
Yeah, I think the assignment of the instance variable `@releases` was making me question it